timelock-impl/src/main/java/com/palantir/atlasdb/timelock/auth/config/ClientPrivilegesConfiguration.java

```diff
@@ -35,15 +39,29 @@
 public abstract class ClientPrivilegesConfiguration implements PrivilegesConfiguration {
     static final String TYPE = "client";
 
+    private static final List<String> BANNED_SUFFIXES = ImmutableList.of("", " ");
+
     @JsonProperty("client-id")
     @Override
     public abstract ClientId clientId();
 
     abstract Set<TimelockNamespace> namespaces();
 
+    @JsonProperty("namespace-suffixes")
+    @Value.Default
+    public Set<String> namespaceSuffixes() {
+        return ImmutableSet.of();
+    }
+
     @Value.Derived
     @Override
     public Privileges privileges() {
-        return NamespacePrivileges.of(namespaces());
+        return NamespacePrivileges.of(namespaces(), namespaceSuffixes());
+    }
+
+    @Value.Check
+    protected void check() {
+        BANNED_SUFFIXES.forEach(bannedSuffix -> Preconditions.checkArgument(!namespaceSuffixes().contains(bannedSuffix),
```

Comment:
I don't believe this check is sufficient. You want to `trim()` and then check it's non-empty + you also want to validate that the suffix would be an acceptable timelock namespace. Something along the lines of this:

Make `TimelockSuffix` an immutable type or call it something better. See [here](https://immutables.github.io/immutable.html#normalization) for immutables normalisation usage.

```java
@Value.Check
protected TimelockSuffix check() {
  if (!value().trim().equals(value()) {
    return TimelockSuffix.of(value().trim());
  }

  Preconditions.checkArgument(isValidClient(value(), "suffix is not a valid client");
  return this;
}
```

Reply:
alternatively, it'd be sufficient to just check that the `value()`  is a valid namespace, instead of trying to normalise it, this is the same as a normal check.